spec/requests/cors_spec.rb

```diff
@@ -0,0 +1,53 @@
+require 'rails_helper'
+require 'geocoding'
+
+RSpec.describe 'CORS', type: :request do
+  describe '/api/v1/coordinates/:location.json' do
+    let(:location_query) { 'ct9' }
+
+    scenario 'does not allow a request from a different domain' do
+      params = { api_version: 1, location: location_query, format: 'json' }
+      headers = { 'HTTP_ORIGIN': 'https://www.test.com' }
+      get api_path(params), headers: headers
+
+      expect(response).to have_http_status(:not_found)
+    end
+
+    scenario 'allows a request from a domain defined by configuration' do
+      geocoding = Geocoding.new(location_query)
+      allow(Geocoding).to receive(:new).with(location_query).and_return(geocoding)
+      allow(geocoding).to receive(:coordinates).and_return([0, 0])
+
+      params = { api_version: 1, location: location_query, format: 'json' }
+      headers = { 'HTTP_ORIGIN': Rails.application.config.allowed_cors_origin }
+      get api_path(params), headers: headers
+
+      expect(response).to have_http_status(:ok)
+      expect(response.headers['Access-Control-Allow-Origin']).to eq(Rails.application.config.allowed_cors_origin)
+    end
+  end
+
+  describe '/api/v1/jobs.json' do
+    scenario 'allows a request from a different domain' do
```

Comment:
I'm a bit confused by this.  Do you mean it is not under CORS control and will allow requests from any domain, or do you mean that it allows requests from a different domain than it is being served from because it is configured to do so?

Reply:
I renamed the test. It's under CORS control which specifically configures it to allow requests from any domain ('*').